corehq/apps/app_manager/views/download.py

```diff
@@ -283,16 +284,19 @@ def create_build_files_if_necessary_handling_conflicts(is_retry=False):
             request.app = Application.get(request.app.get_id)
             create_build_files_if_necessary_handling_conflicts(True)
 
+    # Todo; remove after https://dimagi-dev.atlassian.net/browse/ICDS-1483 is fixed
+    extension = path.split(".")[-1]
+    if extension not in content_type_map.keys():
+        metrics_counter("commcare.invalid_download_requests",
```

Comment:
@orangejenny Added this to track unexpected paths to see if they are all invalid so that we can 404 earlier.

Reply:
clever 👍

Reply:
So far `mp3, mp4, 3gp, png` and no xml files. So I guess we can go ahead and safely 404 on this without building the app files. @snopoke